src/libasr/asr_verify.cpp

```diff
@@ -440,6 +440,25 @@ class VerifyVisitor : public BaseWalkVisitor<VerifyVisitor>
         verify_unique_dependencies(x.m_dependencies, x.n_dependencies,
                                    x.m_name, x.base.base.loc);
 
+        // Get the x symtab.
+        SymbolTable *sym = x.m_symtab->parent;
+
+        // Dependencies of the function should be from function's symbol table.
+        for( size_t i = 0; i < x.n_dependencies; i++ ) {
+            std::string found_dep = x.m_dependencies[i];
+
+            // Get the symbol of the found_dep.
+            ASR::symbol_t* dep_sym = sym->get_symbol(found_dep);
+
+            // If dep_sym is External Symbol or nullptr. It should be skipped.
+            if (dep_sym == nullptr || ASR::is_a<ASR::ExternalSymbol_t>(*dep_sym)) {
+                continue;
+            }  
+
+            require(dep_sym != nullptr,
+                    "Dependency " + found_dep +  " does not belong to same parent symbol table of " + std::string(x.m_name));
+        }
```

Comment:
I am trying to understand the logic here.

Are we ensuring that all dependencies are from the same parent symbol table?

It seems this logic is testing that all the symbols (except ExternalSymbol) are NOT from the parent symbol table. Tests pass, so I guess this is the correct logic, but I don't understand it. Once I understand it, let's document this  as a comment.